src/Network/Skylark/Core/Types.hs

```diff
@@ -217,85 +214,83 @@ instance Var LogLevel where
+  confLogLevel  = cId . lens _confLogLevel (\s a -> s { _confLogLevel = a } )
 
-instance HasConfig Config where
+instance HasConf Conf where
   cId = id
 
-instance Default Config where
-  def = Config
-    { _cConfigFile = Just "/conf/dev.yaml"
-    , _cPort       = Just 3030
-    , _cTimeout    = Just 120
-    , _cLogLevel   = Just LevelInfo
+instance Default Conf where
+  def = Conf
+    { _confFile     = Just "/conf/dev.yaml"
+    , _confPort     = Just 5000
+    , _confTimeout  = Just 120
+    , _confLogLevel = Just LevelInfo
     }
 
-instance FromJSON Config where
+instance FromJSON Conf where
   parseJSON (Object v) =
-    Config                 <$>
-      v .:? "conf-file"    <*>
-      v .:? "port"         <*>
-      v .:? "timeout"      <*>
+    Conf                <$>
+      v .:? "conf-file" <*>
+      v .:? "port"      <*>
+      v .:? "timeout"   <*>
       v .:? "log-level"
   parseJSON _ = mzero
 
-instance FromEnv Config where
+instance FromEnv Conf where
   fromEnv =
-    Config                        <$>
-      envMaybe "SKYLARK_CONFFILE" <*>
-      envMaybe "SKYLARK_PORT"     <*>
-      envMaybe "SKYLARK_TIMEOUT"  <*>
-      envMaybe "SKYLARK_LOGLEVEL"
-
-instance ToEnv Config where
-  toEnv Config{..} =
-    makeEnv [ "SKYLARK_CONFFILE" .= _cConfigFile
-            , "SKYLARK_PORT"     .= _cPort
-            , "SKYLARK_TIMEOUT"  .= _cTimeout
-            , "SKYLARK_LOGLEVEL" .= _cLogLevel
-            ]
-
-instance Monoid Config where
-  mempty = Config
-    { _cConfigFile    = Nothing
-    , _cPort          = Nothing
-    , _cTimeout       = Nothing
-    , _cLogLevel      = Nothing
-    }
-
-  a `mappend` b = Config
-    { _cConfigFile    = merge _cConfigFile a b
-    , _cPort          = merge _cPort a b
-    , _cTimeout       = merge _cTimeout a b
-    , _cLogLevel      = merge _cLogLevel a b
+    Conf                           <$>
+      envMaybe "SKYLARK_CONF_FILE" <*>
+      envMaybe "SKYLARK_PORT"      <*>
+      envMaybe "SKYLARK_TIMEOUT"   <*>
+      envMaybe "SKYLARK_LOG_LEVEL"
+
+instance ToEnv Conf where
+  toEnv Conf{..} = makeEnv
+    [ "SKYLARK_CONF_FILE" .= _confFile
+    , "SKYLARK_PORT"      .= _confPort
+    , "SKYLARK_TIMEOUT"   .= _confTimeout
+    , "SKYLARK_LOG_LEVEL" .= _confLogLevel
+    ]
+
+instance Monoid Conf where
+  mempty = Conf
+    { _confFile     = Nothing
+    , _confPort     = Nothing
+    , _confTimeout  = Nothing
+    , _confLogLevel = Nothing
     }
 
--- | Given a record field accessor. return the second non-Nothing
--- Value for a record field.
---
-merge :: forall a a1. (a1 -> Maybe a) -> a1 -> a1 -> Maybe a
-merge f a b = getLast $! (mappend `on` (Last . f)) a b
+  mappend a b = Conf
+    { _confFile     = merge _confFile a b
+    , _confPort     = merge _confPort a b
+    , _confTimeout  = merge _confTimeout a b
+    , _confLogLevel = merge _confLogLevel a b
+    } where
+      -- | Given a record field accessor. return the second non-Nothing
```

Comment:
Hmmm, I think we'd still have to define a Monoid type for each new instance of `HasConf`. In which case, we should export a common definition of `merge`.